### DISCUSSION TITLE
Ways to update pkg.go.dev cache

### DISCUSSION THREAD

**Discussion by jihoon-seo** [cloud-barista/cb-tumblebug#390] [General]
현재 https://pkg.go.dev/github.com/cloud-barista/cb-tumblebug 페이지에 방문하면
pkg.go.dev 에서 최신 tag 라고 인식한 [`v0.2.9`](https://github.com/cloud-barista/cb-tumblebug/releases/tag/v0.2.9) 의 코드 형상 (https://github.com/cloud-barista/cb-tumblebug/commit/35e14a7c2542430e15b22cb56cd536048594fe1b) 의 godoc comments 를 수집하여
godoc 형태로 보여 주고 있습니다.

## [[방법 A](https://github.com/golang/go/issues/37106)]
https://pkg.go.dev/github.com/cloud-barista/cb-tumblebug@master 주소를 웹 브라우저로 방문하면
`master` 브랜치의 **최신** 코드 형상에 따른 godoc 을 출력합니다. (+ 필요한 경우 pkg.go.dev 캐시도 업데이트)

## [[특정 tag 에 대해 pkg.go.dev 캐시를 업데이트 하는 방법 B](https://stackoverflow.com/questions/60217151/how-to-release-updated-packages-for-go-mod-pkg-go-dev-consumers)]
`curl https://sum.golang.org/lookup/github.com/cloud-barista/cb-tumblebug@v0.3.0-espresso`

## [[특정 tag 에 대해 pkg.go.dev 캐시를 업데이트 하는 방법 C](https://dev.to/koddr/how-to-update-version-s-cache-of-your-package-in-pkg-go-dev-39ij)]
[in Makefile]
```Makefile
update-pkg-cache:
    GOPROXY=https://proxy.golang.org GO111MODULE=on \
    go get github.com/$(USER)/$(PACKAGE)@$(VERSION)
```

[and use like this:]
```Shell
make update-pkg-cache USER=cloud-barista PACKAGE=cb-tumblebug VERSION=v0.3.0-espresso
```

**Comment by powerkimhub** [comment]
## [방법 C: 응용]
[W/O make:]
```Shell
$ GOPROXY=https://proxy.golang.org GO111MODULE=on go get github.com/cloud-barista/cb-spider@master
$ GOPROXY=https://proxy.golang.org GO111MODULE=on go get github.com/cloud-barista/cb-store@v0.3.13
```

**Comment by yunkon-kim** [comment]
[참고자료]
"pkg.go.dev cache"에 대한 이해를 돕기위한 단순 참고자료 입니다 :paw_prints:
> Hi, DEV people! 😉 Sometimes, when you publish a new version of your Go package, pkg.go.dev may still give away an old version for a long time. - [How to update version's cache of your package in pkg.go.dev?](https://dev.to/koddr/how-to-update-version-s-cache-of-your-package-in-pkg-go-dev-39ij)

**Comment by powerkimhub** [comment]
[사례] 신규 태깅 버전 pkg.go.dev 웹페이지 반영 관련
1. now Tag: v0.3.11
2. new Tagging: v0.3.12
3. C방법 캐시 업데이트
  - 대략 8시간 경과
  - pkg.go.dev 웹페이지: 그림과 같이, v0.3.12 docs 생성 후 "GO TO LATEST" 버튼으로 표시만 <br>

     ![image](https://user-images.githubusercontent.com/46367962/107895500-ee505c00-6f76-11eb-8980-45c37647856e.png)

  - A방법: https://pkg.go.dev/github.com/cloud-barista/cb-spider@master 방문
    - 동일하게 "GO TO LATEST" 버튼으로 표시만

4. A방법 응용: https://pkg.go.dev/github.com/cloud-barista/cb-spider 방문
  - pkg.go.dev 웹페이지: v0.3.12가 default 페이지로 보임 <br>

     ![image](https://user-images.githubusercontent.com/46367962/107895706-a978f500-6f77-11eb-87d4-a5017d1576ad.png)
